### ISSUE TITLE
Build phase fails with "No such file or directory"

### ISSUE THREAD

**Issue by garciabruno** [getsentry/sentry-react-native#2722]
**OS:**
- [ ]  Windows
- [X]  MacOS
- [ ]  Linux

**Platform:**
- [X]  iOS
- [ ]  Android

**SDK:**
- [X]  `@sentry/react-native` (>= 1.0.0)
- [ ]  `react-native-sentry` (<= 0.43.2)

**SDK version:** 4.12.0

**`react-native` version:** 0.70.6

**Are you using Expo?**
- [ ]  Yes
- [X]  No

**Are you using sentry.io or on-premise?**
- [X]  sentry.io (SaaS)
- [ ]  on-premise

**Configuration:**

(`@sentry/react-native`)
```
Sentry.init({
  dsn: '[REDACTED]'
});
```

---
**I have the following issue:**

I'm trying to install Sentry SDK integration on iOS, on a MacBook Air (with M1 chip), but the build fails during the "Bundle React Native code and images" build phase.

I looked at other issues reported here with a similar situation than mine (#2067, #484), but no solution was found for me.

Furthermore, I am **not** using NVM, I installed node from Homebrew.
My node version is 19.1.0
I am using Kitty (zsh) as my shell, however, even though I changed the binary on the Build phase to `zsh`, it still fails.

The issue in question is very odd, because I ran the following command before the `sentry-cli` command during that particular Build phase, and it seems like the shell doesn't have access, or cannot find the `ios` directory from which is running at.

```
echo "-------------------------"
echo "Building React Native code and images at path $(pwd)"
echo "-------------------------"
echo "$("ls $(pwd)")"

set -e

WITH_ENVIRONMENT="../node_modules/react-native/scripts/xcode/with-environment.sh"
sentry-cli react-native xcode REACT_NATIVE_XCODE="$(pwd)/../node_modules/react-native/scripts/react-native-xcode.sh"

(...)
```

Which produced the following output:

```
-------------------------
Building React Native code and images at path /Users/[REDACTED]/src/[REDACTED]/ios
-------------------------
/Users/[REDACTED]/Library/Developer/Xcode/DerivedData/[REDACTED]-bvqxlakicoryckfugdirqmwbyzol/Build/Intermediates.noindex/[REDACTED].build/Debug-iphonesimulator/[REDACTED].build/Script-00DD1BFF1BD5951E006B06BC.sh: line 8: ls /Users/[REDACTED]/src/[REDACTED]/ios: No such file or directory

  INFO    2022-12-28 21:40:12.420966 -03:00 Loaded file referenced by SENTRY_PROPERTIES (sentry.properties)
  DEBUG   2022-12-28 21:40:12.421464 -03:00 sentry-cli version: 2.11.0, platform: "darwin", architecture: "arm64"
  INFO    2022-12-28 21:40:12.421471 -03:00 sentry-cli was invoked with the following command line: "sentry-cli" "react-native" "xcode" "REACT_NATIVE_XCODE=/Users/[REDACTED]/src/[REDACTED]/ios/../node_modules/react-native/scripts/react-native-xcode.sh"
error: No such file or directory (os error 2)
  DEBUG   2022-12-28 21:40:12.421551 -03:00 skipping update nagger because session is not attended

GenerateDSYMFile /Users/[REDACTED]/Library/Developer/Xcode/DerivedData/[REDACTED]-bvqxlakicoryckfugdirqmwbyzol/Build/Products/Debug-iphonesimulator/[REDACTED].app.dSYM /Users/[REDACTED]/Library/Developer/Xcode/DerivedData/[REDACTED]-bvqxlakicoryckfugdirqmwbyzol/Build/Products/Debug-iphonesimulator/[REDACTED].app/[REDACTED] (in target '[REDACTED]' from project '[REDACTED]')
    cd /Users/[REDACTED]/src/[REDACTED]/ios
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/dsymutil /Users/[REDACTED]/Library/Developer/Xcode/DerivedData/[REDACTED]-bvqxlakicoryckfugdirqmwbyzol/Build/Products/Debug-iphonesimulator/[REDACTED].app/[REDACTED] -o /Users/[REDACTED]/Library/Developer/Xcode/DerivedData/[REDACTED]-bvqxlakicoryckfugdirqmwbyzol/Build/Products/Debug-iphonesimulator/[REDACTED].app.dSYM


** BUILD FAILED **
```

The build phase cannot even access the `ios` directory, nor the node_modules directory, which contains the `react-native-xcode.sh` file it's expecting.

**Steps to reproduce:**
- Installed @sentry/react-native through yarn in my Ubuntu environment
- Ran the wizard for both platforms (iOS and Android) in the Ubuntu environment, the wizard patched all files for both platforms correctly.
- On Android, everything built and ran correctly
- Proceeded to switch to my iOS environment, installed packages, ran `pod install` and then proceeded to build the app with `npx react-native run-ios`
- Build failed at the Build phase step "Bundle React Native code and images" with a `No such file or directory` error

**Actual result:**

(See log above)

**Expected result:**

App to build on iOS

---

Any help is appreciated

**Comment by garciabruno** [comment]
I tried reproducing the error in a blank application created through the react-native cli from macOS, and the error is the same.

**Comment by krystofwoldrich** [comment]
The `ls` command fails because of the extra quotes.

```diff
- echo "$("ls $(pwd)")"
+ echo "$(ls $(pwd))"
```

The`sentry-cli` command doesn't look right. It must be executed with the environment and has a path to the `react-native-xcode.sh` as the last argument. More [in the docs](https://docs.sentry.io/platforms/react-native/troubleshooting/#react-native-069-and-higher).

```diff
- WITH_ENVIRONMENT="../node_modules/react-native/scripts/xcode/with-environment.sh"
- sentry-cli react-native xcode REACT_NATIVE_XCODE="$(pwd)/../node_modules/react-native/scripts/react-native-xcode.sh"
+ WITH_ENVIRONMENT="../node_modules/react-native/scripts/xcode/with-environment.sh"
+ REACT_NATIVE_XCODE="../node_modules/react-native/scripts/react-native-xcode.sh"
+
+ /bin/sh -c "$WITH_ENVIRONMENT \\"../node_modules/@sentry/cli/bin/sentry-cli react-native xcode $REACT_NATIVE_XCODE\\""
```

Did you add this code manually or was it generated like this by the `@sentry/wizard`?

Let us know if the proposed changes fixed the issue.

**Comment by garciabruno** [comment]
Hey, thanks for your reply !

The code was indeed generated by `@sentry/wizard`.
You were right about the quotes on the ls command, and your modifications made to the script solved the issue.
Some small details:

1. I had to create a `.xcode.env` file in the `ios` directory with my `node` binary path as follows:

```
export NODE_BINARY="/opt/homebrew/bin/node"
```
(Or the equivalent result of `which node`)

2. I had to `brew install coreutils` in order to get the `realpath` binary

3. I made a small modification to the script, I'm not sure if it was failing due to quotes or the previously mentioned items, but the final version working for me is this:

```
export SENTRY_PROPERTIES=sentry.properties
export EXTRA_PACKAGER_ARGS="--sourcemap-output $DERIVED_FILE_DIR/main.jsbundle.map"

set -e

WITH_ENVIRONMENT="../node_modules/react-native/scripts/xcode/with-environment.sh"
REACT_NATIVE_XCODE="../node_modules/react-native/scripts/react-native-xcode.sh"
SENTRY_CLI_XCODE="../node_modules/@sentry/cli/bin/sentry-cli react-native xcode $REACT_NATIVE_XCODE"

/bin/sh -c "$WITH_ENVIRONMENT \"${SENTRY_CLI_XCODE}\""
```

Again, thank you so much, and you can mark this issue as solved.

**Comment by krystofwoldrich** [comment]
Thanks for the details. I'm happy that it worked.

**Comment by walterholohan** [comment]
Thank you, adding the below really helped for me
```
export SENTRY_PROPERTIES=sentry.properties
export EXTRA_PACKAGER_ARGS="--sourcemap-output $DERIVED_FILE_DIR/main.jsbundle.map"

set -e

WITH_ENVIRONMENT="../node_modules/react-native/scripts/xcode/with-environment.sh"
REACT_NATIVE_XCODE="../node_modules/react-native/scripts/react-native-xcode.sh"
SENTRY_CLI_XCODE="../node_modules/@sentry/cli/bin/sentry-cli react-native xcode $REACT_NATIVE_XCODE"

/bin/sh -c "$WITH_ENVIRONMENT \"${SENTRY_CLI_XCODE}\""
```

**Comment by imdivyang** [comment]
@walterholohan is not working for me.please help if any other solution is their.

**Comment by Peege151** [comment]
I had this issue as well.  FWIW, I was given the warning on react-native run-ios --device 

```
Package @sentry/react-native contains invalid configuration: "dependency.platforms.ios.sharedLibraries" is not allowed,"dependency.hooks" is not allowed. Please verify it's properly linked using "react-native config" command and contact the package maintainers about this.
```
If we get this issue is there something different about our environment?

Will the sentry wizard be patched?

**Comment by Peege151** [comment]
After uninstalling, I'm still getting this.

```
    /bin/sh -c /Users/patrick/Library/Developer/Xcode/DerivedData/AwesomeProject-cbtuwiwmzvdegzfqqvdbqodlnixq/Build/Intermediates.noindex/AwesomeProject.build/Debug-iphoneos/AwesomeProject.build/Script-00DD1BFF1BD5951E006B06BC.sh
/Users/patrick/Library/Developer/Xcode/DerivedData/AwesomeProject-cbtuwiwmzvdegzfqqvdbqodlnixq/Build/Intermediates.noindex/AwesomeProject.build/Debug-iphoneos/AwesomeProject.build/Script-00DD1BFF1BD5951E006B06BC.sh: line 6: ../node_modules/@sentry/cli/bin/sentry-cli: No such file or directory
Command PhaseScriptExecution failed with a nonzero exit code
```
I've deleted derived data, and the build... this happens every build. I've uninstalled the pod, and npm modules (which is why it's complaining) but there's something still referencing..

**Comment by krystofwoldrich** [comment]
@ziyoshams I'm sorry for the trouble, here are the [fixed scripts](https://github.com/getsentry/sentry-docs/pull/6178).

**Comment by krystofwoldrich** [comment]
@imdivyang Do you have the same error or similar, maybe you can open a new issue with your error.

**Comment by krystofwoldrich** [comment]
@Peege151 For the build error, you can find the fix [here](https://docs.sentry.io/platforms/react-native/troubleshooting/#react-native-069-and-higher).

And guide on how to uninstall `@sentry/react-native`, you can find [here](https://docs.sentry.io/platforms/react-native/uninstall/).

**Comment by mtnt** [comment]
so, why it is still not released? v4.15.0 still adds
`../node_modules/@sentry/cli/bin/sentry-cli react-native xcode REACT_NATIVE_XCODE="../node_modules/react-native/scripts/react-native-xcode.sh`

without `\n'

**Comment by krystofwoldrich** [comment]
@mtnt `v4` uses `@sentry/wizard@1` which doesn't support `react-native@>=0.69.0`.
This is fixed in `v5` which uses `@sentry/wizard@2` and patches the files correctly.
